madara/crates/client/sync/src/l2.rs

```diff
@@ -224,6 +226,51 @@ async fn l2_pending_block_task(
     Ok(())
 }
 
+async fn l2_highest_block_fetch(
+    provider: Arc<GatewayProvider>,
+    mut ctx: ServiceContext,
+    sync_status_provider: Arc<SyncStatusProvider>,
+) -> anyhow::Result<()> {
+    tracing::debug!("Start highest block poll");
+
+    let mut interval = tokio::time::interval(Duration::from_secs(10));
+    interval.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
+
+    // Keep running until cancelled
+    while ctx.run_until_cancelled(interval.tick()).await.is_some() {
+        tracing::debug!("Getting highest block...");
+
+        // Try to get the latest block
+        match provider.get_block(BlockId::Tag(BlockTag::Latest)).await {
+            Ok(block) => {
+                match &block {
+                    ProviderBlockPendingMaybe::NonPending(block_info) => {
+                        // For a regular block, we have both number and hash
+                        let block_number = block_info.block_number;
+                        let block_hash = block_info.block_hash;
+
+                        // Update the sync status provider with the highest block info
+                        sync_status_provider.set_highest_block_num(block_number).await;
+                        sync_status_provider.set_highest_block_hash(block_hash).await;
+
+                        tracing::debug!("Updated highest block: number={}, hash={}", block_number, block_hash);
+                    }
+                    _ => {
+                        tracing::warn!("Got unexpected block type from provider");
+                        continue;
```

Comment:
This seems like it could be problematic. In a best case scenario it would skip blocks we don't understand, and in a worst case scenario I could see it creating an infinite loop.

I would suggest failing hard in this case or putting in some mechanism to prevent it from getting stuck.